JSTests/wasm/gc/array_new_fixed.js

```diff
@@ -0,0 +1,599 @@
+    (call $set_get_array (local.get $i)
+      (call $new)
+      (array.new_canon_default $bvec (local.get $len)))
+  )
+
+  (func (export "set_get_byte") (param $i i32) (param $j i32) (param $value i32) (result i32)
+    (call $set_get_byte (local.get $i)
+      (call $new)
+      (local.get $j)
+      (local.get $value))
+  )
+
+  (func $len (param $v (ref array)) (result i32)
+    (array.len (local.get $v))
+  )
+  (func (export "len") (result i32)
+    (call $len (call $new))
+  ))`);
+
+    for (var i = 0; i < 2; i++) {
+        for (var j = 0; j < (i === 0 ? 5 : 3); j++) {
+            assert.eq(m.exports.set_get_byte(i, j, 127), 127);
+        }
+    }
+
+    for (var i = 0; i < 2; i++) {
+        assert.eq(m.exports.set_get_array(i, 12), 12);
+    }
+
+    assert.eq(m.exports.len(), 2);
+    assert.throws(() => m.exports.set_get_array(10, 1), WebAssembly.RuntimeError, "Out of bounds array.set");
+    assert.throws(() => m.exports.set_get_byte(10, 1, 42), WebAssembly.RuntimeError, "Out of bounds array.get");
+    assert.throws(() => m.exports.set_get_byte(1, 4, 42), WebAssembly.RuntimeError, "Out of bounds array.set");
+    assert.throws(() => m.exports.set_get_byte(0, 6, 42), WebAssembly.RuntimeError, "Out of bounds array.set");
+}
+
+function testMissingArgumentCount() {
+    /*
+(module
+  (type $vec (array f32))
+
+  (func $new (export "new") (result (ref $vec))
+    (array.new_canon_fixed $vec)
+  ))
+     */
+
+    assert.throws(() => new WebAssembly.Instance(module("\x00\x61\x73\x6D\x01\x00\x00\x00\x01\x89\x80\x80\x80\x00\x02\x5E\x7D\x00\x60\x00\x01\x6B\x00\x03\x82\x80\x80\x80\x00\x01\x01\x07\x87\x80\x80\x80\x00\x01\x03\x6E\x65\x77\x00\x00\x0A\x91\x80\x80\x80\x00\x01\x8A\x80\x80\x80\x00\x00\x43\x00\x00\x80\x3F\xFB\x19\x00\x0B")), WebAssembly.CompileError, "WebAssembly.Module doesn't parse at byte 56: parsing ended before the end of Code section");
+
+    /*
+(module
+  (type $vec (array f32))
+
+  (func $new (export "new") (result (ref $vec))
+    (array.new_canon_fixed $vec <invalid i32>)
+  ))
+*/
+    assert.throws(() => new WebAssembly.Instance(module("\x00\x61\x73\x6D\x01\x00\x00\x00\x01\x89\x80\x80\x80\x00\x02\x5E\x7D\x00\x60\x00\x01\x6B\x00\x03\x82\x80\x80\x80\x00\x01\x01\x07\x87\x80\x80\x80\x00\x01\x03\x6E\x65\x77\x00\x00\x0A\x91\x80\x80\x80\x00\x01\x8B\x80\x80\x80\x00\x00\x43\x00\x00\x80\x3F\xFB\x19\x00\x99\x99")), WebAssembly.CompileError, "WebAssembly.Module doesn't parse at byte 11: can't get argument count for array.new_fixed, in function at index 0");
+}
+
+
+testArrayNewFixedNested();
+testArrayNewFixed();
+testArrayNewFixedImmutable();
+testZeroLengthArray();
+testSingletonArray()
+testTypeMismatch();
+testArgLengthMismatch();
+testInvalidArrayType();
+testSubtyping();
+
+testPackedTypes();
+testNumericTypes();
+testNestedArrays();
+// FIXME: Uncomment when https://bugs.webkit.org/show_bug.cgi?id=252538
+// and https://bugs.webkit.org/show_bug.cgi?id=252719 are fixed
+// testStructs();
```

Comment:
I wanted to uncomment this line now that these bugs are both fixed, but I found that the test fails if I do; I'm working on it, converted the PR to draft in the meantime.

Reply:
See https://bugs.webkit.org/show_bug.cgi?id=254226